scripts/auto-rebase/rebase_gateway_api.sh

```diff
@@ -0,0 +1,470 @@
+    fi
+
+    title "# Removing staging directory"
+    rm -rf "${STAGING_DIR}"
+}
+
+# OSSM is not integrated into the ocp release image, so the workflow does not fit with core component rebase.
+# Because OSSM is used without OLM, all of the manifests in the bundle are downloaded.
+download_ossm_operator_bundle_manifest() {
+  bundle_manifest="$1"
+  gateway_api_version="$2"
+
+  title "downloading OSSM operator bundle manifests ${bundle_manifest}"
+  local OSSM_STAGING="${STAGING_DIR}/ossm"
+  rm -rf "${OSSM_STAGING}"
+  mkdir -p "${OSSM_STAGING}"
+
+  # Persist the version of the OSSM operator bundle for use in manifest steps
+  local version
+  version=$(echo "${bundle_manifest}" | awk -F':' '{print $2}')
+
+  title "recognized version: ${version}"
+
+  authentication=""
+  if [ -f "${PULL_SECRET_FILE}" ]; then
+    authentication="--registry-config ${PULL_SECRET_FILE}"
+  else
+    >&2 echo "Warning: no pull secret found at ${PULL_SECRET_FILE}"
+  fi
+
+  local -r csv="servicemeshoperator3.clusterserviceversion.yaml"
+  local -r namespace=$(yq '.metadata.name' "${REPOROOT}/assets/optional/gateway-api/00-openshift_gateway_api_namespace.yaml")
+
+  for arch in "${ARCHS[@]}"; do
+    mkdir -p "${OSSM_STAGING}/${arch}"
+    pushd "${OSSM_STAGING}/${arch}" || return 1
+    title "extracting OSSM operator bundle for \"${arch}\" architecture"
+    # shellcheck disable=SC2086  # Double quote to prevent globbing and word splitting.
+    oc image extract \
+      ${authentication} \
+      --path /manifests/:. "${bundle_manifest}" \
+      --filter-by-os "${arch}" \
+      ||  {
+        popd
+        return 1
+      }
+
+      download_gateway_api_manifests "${PWD}" "${gateway_api_version}"
+      extract_ossm_rbac_from_cluster_service_version "${PWD}" "${csv}" "${namespace}"
+      extract_ossm_deploy_from_cluster_service_version "${PWD}" "${csv}" "${namespace}"
+
+      for file in "${PWD}"/*; do
+        if [[ ${file} == *.yaml || ${file} == *.yml ]]; then
+            patch_namespace "Service" "${file}" "${namespace}"
+            patch_namespace "Deployment" "${file}" "${namespace}"
+            patch_image_values "Deployment" "${file}"
+        fi
+      done
+
+      popd || return 1
+    done
+}
+
+patch_namespace() {
+  local kind=$1
+  local file=$2
+  local namespace=$3
+
+  if [[ $(yq e ".kind == \"${kind}\"" "${file}") == "true" ]]; then
+    # Check if the .metadata.namespace is not set or empty
+    if [[ $(yq e ".metadata.namespace == \"${namespace}\"" "${file}") == "false" ]]; then
+      echo "patching .metadata.namespace to \"${namespace}\" in ${file}"
+      # Set the .metadata.namespace to the specified value
+      ns=${namespace} yq e '.metadata.namespace = strenv(ns)'  -i "${file}"
+    fi
+  fi
+}
+
+patch_image_values() {
+  local kind=$1
+  local file=$2
+
+  if [[ $(yq e ".kind == \"${kind}\"" "${file}") == "true" ]]; then
+    yq 'with(.spec.template.spec.containers[]; .image=.name)' -i "${file}"
```

Comment:
I think we generally place `-i` right after `yq` so it's more visible it's editing the file

Reply:
Done